packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -142,7 +142,13 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
     const superProto = getCtorProto(Ctor);
     const superDef =
         superProto !== LightningElement ? getComponentInternalDef(superProto) : lightingElementDef;
-    const bridge = HTMLBridgeElementFactory(superDef.bridge, keys(apiFields), keys(apiMethods));
+    const bridge = HTMLBridgeElementFactory(
+        superDef.bridge,
+        keys(apiFields),
+        keys(apiMethods),
+        keys(observedFields),
```

Comment:
This adds an additional `keys()` in prod mode, but I can't imagine it's a huge perf hit.